### DISCUSSION TITLE
Missing class defintion in app.css

### DISCUSSION THREAD

**Discussion by vv-ac** [filamentphp/filament#4252] [Help]
Hi,

just set up a new Filament installation and did the following:
**Installed dependencies:**
npm install tailwindcss @tailwindcss/forms @tailwindcss/typography autoprefixer tippy.js --save-dev

**Initiated tailwind.**

**tailwind.config.js**
```
/** @type {import('tailwindcss').Config} */

const colors = require('tailwindcss/colors')

module.exports = {
  content: [
    './resources/**/*.blade.php',
    './resources/**/*.js',
    './resources/**/*.vue',
    './resources/**/*.blade.php',
    './vendor/filament/**/*.blade.php', 
    "./app/Filament/Resources/AssetResource.php",
  ],
  darkMode: 'class',
  theme: {
      extend: {
          colors: {
              danger: colors.rose,
              primary: colors.emerald,
              secondary: colors.slate,
              success: colors.green,
              warning: colors.yellow,
              cyan: colors.cyan,
              indigo: colors.indigo,
          },
      },
  },
  plugins: [
    require('@tailwindcss/forms'), 
    require('@tailwindcss/typography'),
  ],
}
```

**app.css**
```
@import '../../vendor/filament/forms/dist/module.esm.css';
 
@tailwind base;
@tailwind components;
@tailwind utilities;
```

**vite.config.js**
```
import { defineConfig } from 'vite';
import laravel, { refreshPaths } from 'laravel-vite-plugin'; 
 
export default defineConfig({
    plugins: [
        laravel({
            input: [
                'resources/css/app.css',
                'resources/js/app.js',
            ],
            refresh: [ 
                ...refreshPaths,
                'app/Http/Livewire/**',
                'app/Tables/Columns/**',
            ], 
        }),
    ],
});
```

**Added a postcss.config.js**
```
module.exports = {
    plugins: {
        tailwindcss: {},
        autoprefixer: {},
    },
}
```

Now, running npm run watch (added it to the scripts) creates the css files w/o any errors (e. g. "public/build/assets/app.9b13d705.css").
Two problems I run into now:
1. In my filament resource files it only adds such classes that I explicitly define like class="bg-indigo-600" but it does not account  for anything in ->color('***') or ->extraAttributes(['class'=>'***']). So, these two are included in the css 
![grafik](https://user-images.githubusercontent.com/67121752/192872500-97781b71-520f-4cf1-991d-3cb30a1f87f1.png)
 but these are ignored 
![grafik](https://user-images.githubusercontent.com/67121752/192872656-8c56d9f9-f644-425e-b7a1-0d3488ffb57b.png)
2. Even though the css definitions are included in the freshly created file it is not included in frontend as part of app.css.

I am clearly missing obvious things here. Can you help me out?

Thanks!

**Comment by danharrin** [comment]
Where are you adding the `extraAttributes()`? Also, `bg-primary` is not valid, should be `bg-primary-600` etc

**Reply by vv-ac** [reply]
Thanks for your reply.
Here's a bigger screenshot:
![grafik](https://user-images.githubusercontent.com/67121752/193083504-b4ee3358-16d8-4e6d-a378-0fa0c04081f8.png)

I must apologize, I actually used another class there and just "faked" it quickly for the purpose of this post with "bg-primary". I tried several valid options which did not lead to success.

Please do not mind the color value, it's just for try and error in there.

**Reply by danharrin** [reply]
I can't reproduce that behaviour... when combining `color()` and `extraAttributes()` on the same column in the latest version, there is only one `class=` attribute.

**Comment by sergioraldana** [comment]
Hi, I have the same issue, did you find the solution?

**Comment by ArtMin96** [comment]
Check this: [PR](https://github.com/filamentphp/filament/pull/5156). You need to set the second parameter to `true` in order to merge and not override.
```php
public function extraAttributes(array | Closure $attributes, bool $merge = false): static
```
